numba/core/inline_closurecall.py

```diff
@@ -262,6 +269,302 @@ def check_reduce_func(func_ir, func_var):
+                items = tuple(cellget(x) for x in closure)
+            else:
+                assert(isinstance(closure, ir.Expr)
+                    and closure.op == 'build_tuple')
+                items = closure.items
+            assert(len(callee_code.co_freevars) == len(items))
+            _replace_freevars(callee_blocks, items)
+        freevars = closure_obj.code.co_freevars
+        return self.inline_ir(caller_ir, block, i, callee_ir, freevars,
+                              arg_typs=arg_typs)
+
+    def run_untyped_passes(self, func):
+        """
+        Run the compiler frontend's untyped passes over the given Python
+        function, and return the function's canonical Numba IR.
+        """
+        from numba.core.compiler import StateDict, _CompileStatus
+        from numba.core.untyped_passes import ExtractByteCode, WithLifting
+        from numba.core import bytecode
+        from numba.parfors.parfor import ParforDiagnostics
+        state = StateDict()
+        state.func_ir = None
+        state.typingctx = self.typingctx
+        state.targetctx = self.targetctx
+        state.locals = self.locals
+        state.pipeline = self.pipeline
+        state.flags = self.flags
+
+        # Disable SSA transformation, the call site won't be in SSA form and
+        # self.inline_ir depends on this being the case.
+        state.flags.enable_ssa = False
+
+        state.func_id = bytecode.FunctionIdentity.from_function(func)
+
+        state.typemap = None
+        state.calltypes = None
+        state.type_annotation = None
+        state.status = _CompileStatus(False, False)
+        state.return_type = None
+        state.parfor_diagnostics = ParforDiagnostics()
+        state.metadata = {}
+
+        ExtractByteCode().run_pass(state)
+        # This is a lie, just need *some* args for the case where an obj mode
+        # with lift is needed
+        state.args = len(state.bc.func_id.pysig.parameters) * (types.pyobject,)
+
+        pm = self._compiler_pipeline(state)
+
+        pm.finalize()
+        pm.run(state)
+        return state.func_ir
+
+    def update_type_and_call_maps(self, callee_ir, arg_typs):
+        """ Updates the type and call maps based on calling callee_ir with arguments
+        from arg_typs"""
+        if not self._permit_update_type_and_call_maps:
+            msg = ("InlineWorker instance not configured correctly, typemap or "
+                   "calltypes missing in initialization.")
+            raise ValueError(msg)
+        from numba.core import typed_passes
+        # call branch pruning to simplify IR and avoid inference errors
+        callee_ir._definitions = ir_utils.build_definitions(callee_ir.blocks)
+        numba.core.analysis.dead_branch_prune(callee_ir, arg_typs)
+        try:
+            f_typemap, f_return_type, f_calltypes = typed_passes.type_inference_stage(
+                    self.typingctx, callee_ir, arg_typs, None)
+        except Exception:
```

Comment:
pretty certain that you don't need this try..except. It's never exercised by the testsuite.

Reply:
I think you are right, I've a feeling this is an artifact of development from the numerous rounds of trying to get typed inlining working.